app/api/event_copy.py

```diff
@@ -36,97 +45,57 @@ def create_event_copy(identifier):
+    discount_codes = DiscountCode.query.filter_by(event_id=event.id, deleted_at=None).all()
+    speaker_calls = SpeakersCall.query.filter_by(event_id=event.id, deleted_at=None).all()
+    user_event_roles = UsersEventsRoles.query.filter_by(event_id=event.id, deleted_at=None).all()
+    taxes = Tax.query.filter_by(event_id=event.id, deleted_at=None).all()
 
     db.session.expunge(event)  # expunge the object from session
     make_transient(event)
     delattr(event, 'id')
     event.identifier = get_new_event_identifier()
     save_to_db(event)
 
+    # Ensure tax information is copied
+    for tax in taxes:
+        expunge_object(tax, event)
+
     # Removes access_codes, order_tickets, ticket_tags for the new tickets created.
     for ticket in tickets:
-        ticket_id = ticket.id
-        db.session.expunge(ticket)  # expunge the object from session
-        make_transient(ticket)
-        ticket.event_id = event.id
-        delattr(ticket, 'id')
-        save_to_db(ticket)
+        expunge_object(ticket, event)
 
     for link in social_links:
-        link_id = link.id
-        db.session.expunge(link)  # expunge the object from session
-        make_transient(link)
-        link.event_id = event.id
-        delattr(link, 'id')
-        save_to_db(link)
+        expunge_object(link, event)
 
     for sponsor in sponsors:
-        sponsor_id = sponsor.id
-        db.session.expunge(sponsor)  # expunge the object from session
-        make_transient(sponsor)
-        sponsor.event_id = event.id
-        delattr(sponsor, 'id')
-        save_to_db(sponsor)
+        expunge_object(sponsor, event)
 
     start_sponsor_logo_generation_task(event.id)
 
     for location in microlocations:
-        location_id = location.id
-        db.session.expunge(location)  # expunge the object from session
-        make_transient(location)
-        location.event_id = event.id
-        delattr(location, 'id')
-        save_to_db(location)
+        expunge_object(location, event)
 
     # No sessions are copied for new tracks
     for track in tracks:
-        track_id = track.id
-        db.session.expunge(track)  # expunge the object from session
-        make_transient(track)
-        track.event_id = event.id
-        delattr(track, 'id')
-        save_to_db(track)
+        expunge_object(track, event)
 
     for call in speaker_calls:
-        call_id = call.id
-        db.session.expunge(call)  # expunge the object from session
-        make_transient(call)
-        call.event_id = event.id
-        delattr(call, 'id')
-        save_to_db(call)
+        expunge_object(call, event)
 
     for code in discount_codes:
-        code_id = code.id
-        db.session.expunge(code)  # expunge the object from session
-        make_transient(code)
-        code.event_id = event.id
-        delattr(code, 'id')
-        save_to_db(code)
+        expunge_object(code, event)
 
     for form in custom_forms:
-        form_id = form.id
```

Comment:
@uds5501 you missed the first step in the function?

Reply:
Which is unneeded. I don't see form_id being used anywhere?